GoRogue.UnitTests/MapGeneration/PolygonAreaTests.cs

```diff
@@ -96,5 +175,188 @@ public void PolygonAlgorithmsMustBeOrderedTest(Lines.Algorithm algorithm)
 
             Assert.Throws<ArgumentException>(() => new PolygonArea(algorithm, p1, p2, p3));
         }
+
+        [Fact]
+        public void PolygonFromRectangle()
+        {
+            var rect = new Rectangle(0, 0, 15, 10);
+            var polygon = PolygonArea.Rectangle(rect);
+
+            //Each Corner is included in two boundaries, so total number of points is offset by 4
+            Assert.Equal(rect.Area + 4, polygon.Count);
+            Assert.Equal(rect.PerimeterPositions().Count() + 4, polygon.OuterPoints.Count);
+            Assert.Equal(new HashSet<Point>(rect.Expand(-1, -1).Positions()), new HashSet<Point>(polygon.InnerPoints));
+            _output.WriteLine(GetPolygonString(polygon));
+        }
+
+        [Fact]
+        public void PolygonParallelogram()
+        {
+            var polygon = PolygonArea.Parallelogram((30, 30), 15, 15);
+            Assert.Equal(4, polygon.Corners.Count);
+            Assert.Equal(196, polygon.InnerPoints.Count);
+            Assert.Equal(64, polygon.OuterPoints.Count);
+            _output.WriteLine(GetPolygonString(polygon));
+        }
+
+        private void AssertPolygonSidesAreEqual(PolygonArea polygon, bool isStar)
+        {
+            var outer = (MultiArea)polygon.OuterPoints;//Satan forgive me
+
+            var allInts = new List<int>();
+            var intsEncountered = new Dictionary<int, int>();
+            int total = 0;
+            var divisor = outer.SubAreas.Count;
+
+            foreach (var boundary in outer.SubAreas)
+            {
+                var count = boundary.Count;
+                total += count;
+                //if we haven't seen it, track it once.
+                if (!intsEncountered.ContainsKey(count))
+                    intsEncountered.Add(count, 1);
+
+                //otherwise, increase the amount of times we've seen it
+                else
+                    intsEncountered[count]++;
+
+                allInts.Add(count);
+            }
+
+            var mean = total / divisor;
+            allInts.Sort();
+            var mode = allInts[allInts.Count / 2];
+            var median = intsEncountered.OrderByDescending(kvp => kvp.Value).First().Key;
+
+            string message = isStar ? "Regular Star" : "Regular Polygon";
+            var cornerCount = isStar ? polygon.Corners.Count / 2 : polygon.Corners.Count;
+
+            message += $" with {cornerCount} corners has a lopside. \r\n";
+            message += $"Mean: {mean}\r\nMedian: {median}\r\nMode: {mode}\r\nHas a side of length: ";
+            foreach (var i in allInts)
```

Comment:
Do we need to use `allInts` here?  `allInts` will contain many duplicate numbers, assuming that many sides contain the same amount of points (which _should_ be the case if the creation functions are functioning properly); however it would appear that we only actually need to run the code in this loop on each unique number once; if that's the case, we could cut down the number of iterations that need to happen significantly by using the keys of `intsEncountered` instead:
```suggestion
            foreach (var i in intsEncountered.Keys)
```